src/orbit-db-identity-provider.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+const IdentityProvider = require('./identity-provider')
+const Keystore = require('orbit-db-keystore')
+const type = 'orbitdb'
+
+class OrbitDBIdentityProvider extends IdentityProvider {
+  constructor(options = {}) {
+    super()
+    this._keystore = options.keystore || Keystore.create(options.keypath || './orbitdb/identity/keys')
+  }
+
+  // Returns the type of the identity provider
+  static get type () { return type }
+
+  async createId(options = {}) {
+    const ipfs = options.ipfs
+    if (!ipfs)
+      throw new Error('ipfs instance required')
+
+    const keystore = options.keystore || this._keystore
+    const { id } = await ipfs.id()
+    this.id = id
+    const key = await keystore.getKey(id) || await keystore.createKey(id)
+    return key.getPublic('hex')
+  }
+
+  async signIdentity(pubKeyIdSig, options = {}) {
+    const keystore = options.keystore || this._keystore
```

Comment:
Is it needed to allow the user to pass the keystore here too (in addition to the constructor)? Is there a use case where the user would use a different keystore for signing than what is passed in the constructor?

Reply:
I think you're right re: which keystore to use is confusing, I think the options arg should remain though as each provider may need sth different passed in to know how to sign (id in this case)

Reply:
We currently pass "something to know how to sign" in the constructor (eg. wallet, ipfs or orbitdb, etc). I think, because "that something" works as the final authority on how/what is signed, it shouldn't be customizable per signature, but should be per provider and passing it in the constructor "fixes it" to that particular provider. That is to say, I don't think the keystore should be customizable per signature and if someone needs that (I'm not aware of any use cases, correct me if I'm out of date) later on, we can still easily add it. As per the previous comment, we should always default to "less" when there's no clear/urgent need for it.